datastream-common/src/main/java/com/linkedin/datastream/common/BrooklinEnvelopeMetadataConstants.java

```diff
@@ -36,4 +36,7 @@
 
   // Timestamp of the event when it was written in the last leg (i.e. the Source of the connector)
   public static final String SOURCE_TIMESTAMP = "SourceTimestamp";
+
+  // Source partition number from where the event was generated
+  public static final String SOURCE_PARTITION = "SourcePartition";
```

Comment:
Nit: Is it possible to merge this constant with "kafka-origin-offset"? It feels like we have some redundancy in metadata fields otherwise.

Reply:
Discussed this a little bit. One problem with reconciling this with "kafka-origin-partition" and even moving all "kafka-origin-*" constants into this file is that these are very specific to Kafka based sources. Want to have a generic field that isn't tied to Kafka, as this concept of "partition" is generic and can even mean a database partitioning scheme. Plus need the constant to be accessible in the transport layers which ideally shouldn't be tied to the source connector type.